File: internal/CalculatorUnitTests/UtilsTests.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include <WexTestClass.h>

namespace CalculatorUnitTests
{
    class UtilsTests
    {
    public:
        TEST_CLASS(UtilsTests);

        TEST_METHOD(IsLastCharacterSuccess)
        {
            VERIFY_IS_TRUE(Utils::IsLastCharacterTarget(L"Test.", L'.'));
        }

        TEST_METHOD(IsLastCharacterSuccessMultipleSuffices)
        {
            VERIFY_IS_TRUE(Utils::IsLastCharacterTarget(L"Test..", L'.'));
        }

        TEST_METHOD(IsLastCharacterFailure)
        {
            VERIFY_IS_FALSE(Utils::IsLastCharacterTarget(L"Test", L'.'));
        }

        TEST_METHOD(IsLastCharacterFailureAllButLastMatch)
        {
            VERIFY_IS_FALSE(Utils::IsLastCharacterTarget(L".....T", L'.'));
        }

        TEST_METHOD(IsLastCharacterFailureEmptyInput)
        {
            VERIFY_IS_FALSE(Utils::IsLastCharacterTarget({}, L'.'));
        }

        TEST_METHOD(IsLastCharacterFailureNullTarget)
        {
            VERIFY_IS_FALSE(Utils::IsLastCharacterTarget({}, NULL));
        }
    };
}

File: src/Calculator.Tests/UtilsTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.VisualStudio.TestTools.UnitTesting;
using CalculatorApp.ViewModel.Common;

namespace Calculator.Tests
{
    [TestClass]
    public class UtilsTests
    {
        [TestMethod]
        public void IsLastCharacterSuccess()
        {
            Assert.IsTrue(Utils.IsLastCharacterTarget("Test.", '.'));
        }

        [TestMethod]
        public void IsLastCharacterSuccessMultipleSuffices()
        {
            Assert.IsTrue(Utils.IsLastCharacterTarget("Test..", '.'));
        }

        [TestMethod]
        public void IsLastCharacterFailure()
        {
            Assert.IsFalse(Utils.IsLastCharacterTarget("Test", '.'));
        }

        [TestMethod]
        public void IsLastCharacterFailureAllButLastMatch()
        {
            Assert.IsFalse(Utils.IsLastCharacterTarget(".....T", '.'));
        }

        [TestMethod]
        public void IsLastCharacterFailureEmptyInput()
        {
            Assert.IsFalse(Utils.IsLastCharacterTarget(string.Empty, '.'));
        }

        [TestMethod]
        public void IsLastCharacterFailureNullTarget()
        {
            Assert.IsFalse(Utils.IsLastCharacterTarget(string.Empty, '\0'));
        }
    }
}
